GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java

```diff
@@ -714,5 +714,27 @@ public void deleteSurveyInstance(SurveyInstance surveyInstance) {
     public SurveyInstance findByUUID(String uuid) {
         return findByProperty("uuid", uuid, "String");
     }
+    
+    public SurveyInstance getRegistrationSurveyInstance(SurveyedLocale locale) {
+        PersistenceManager pm = PersistenceFilter.getManager();
+        Query query = pm.newQuery(SurveyInstance.class);
+
+        Map<String, Object> paramMap = null;
+
+        StringBuilder filterString = new StringBuilder();
+        StringBuilder paramString = new StringBuilder();
+        paramMap = new HashMap<String, Object>();
+
+        appendNonNullParam("surveyedLocaleId", filterString, paramString,
+                "Long", locale.getKey().getId(), paramMap);
+        appendNonNullParam("surveyId", filterString, paramString,
+                "Long", locale.getCreationSurveyId(), paramMap);
+        query.setFilter(filterString.toString());
+        query.declareParameters(paramString.toString());
+        query.setUnique(true);
+        query.setOrdering("collectionDate desc");
```

Comment:
I expect these two filters to make the query more efficient in case we have more than one registration form:

- `setUnique(true)` will make sure only the first result is returned, rather than a Collection, and `null` if it doesn't exist. This way we don't have to fetch a List to check it's first element. Just read the query response in a SurveyInstance variable.

- The ordering one is just to ensure we get the latest registration form, avoiding any subtle surprise we can have if multiple registration forms are present.